api/controllers/reservationController.js

```diff
@@ -0,0 +1,18 @@
+const reservationService = require("../services/reservationService");
+const { catchAsync } = require("../utils/error");
+
+const getOptions = catchAsync(async (req, res) => {
+  const { movies, cinema_names, hall_types, dates } = req.query;
```

Comment:
```suggestion
  const { movie, cinema-name: cinemaName, hall-type: hallType, date } = req.query;
```
query로 매개변수를 받는다면 아래와 같은 형식으로 받게 됩니다. 
```shell
http://127.0.0.1:3000/movies?movie=000&cinema_names=00000
```
REST architecture에서는 uniform interface 원칙에 따르면 언더바 사용을 지양하고 있습니다. 
또한 javascript에서는 마찬가지로 언더바를 사용하는 snake_case를 사용하지 않고 코딩 컨벤션으로 camelCase를 활용합니다. 

따라서 query string을 통해 받는 매겨변수에서 단어의 이음새는 하이픈(`-`)을 사용하고 이를 구조분해할당하면서 이름을 camelCase로 바꿔주면 좋을 것 같습니다.